### ISSUE TITLE
wagtailredirects.middleware causing problems with illegal strings?

### ISSUE THREAD

**Issue by benjaoming** [wagtail/wagtail#4496]
### Issue Summary

A malicious crawl caused this in `wagtailredirects.middleware`:

`A string literal cannot contain NUL (0x00) characters.`

Not sure if this should be the responsibility of `db.backends.utils` or `wagtailredirects`.

### Steps to Reproduce

Add a weird character in the path.

### Technical details

* Python version: 3.5.2.
* Django version: 1.11.12.
* Wagtail version: 1.13.1.
* Browser version: n/a

```

Traceback:  

File ".virtualenvs/venv/lib/python3.5/site-packages/django/core/handlers/base.py" in get_response
  131.                 response = middleware_method(request, response)

File ".virtualenvs/venv/lib/python3.5/site-packages/wagtail/wagtailredirects/middleware.py" in process_response
  51.         redirect = get_redirect(request, path)

File ".virtualenvs/venv/lib/python3.5/site-packages/wagtail/wagtailredirects/middleware.py" in get_redirect
  31.         redirect = _get_redirect(request, uri_to_iri(path))

File ".virtualenvs/venv/lib/python3.5/site-packages/wagtail/wagtailredirects/middleware.py" in _get_redirect
  19.         return models.Redirect.get_for_site(request.site).get(old_path=path)

File ".virtualenvs/venv/lib/python3.5/site-packages/django/db/models/query.py" in get
  374.         num = len(clone)

File ".virtualenvs/venv/lib/python3.5/site-packages/django/db/models/query.py" in __len__
  232.         self._fetch_all()

File ".virtualenvs/venv/lib/python3.5/site-packages/django/db/models/query.py" in _fetch_all
  1118.             self._result_cache = list(self._iterable_class(self))

File ".virtualenvs/venv/lib/python3.5/site-packages/django/db/models/query.py" in __iter__
  53.         results = compiler.execute_sql(chunked_fetch=self.chunked_fetch)

File ".virtualenvs/venv/lib/python3.5/site-packages/django/db/models/sql/compiler.py" in execute_sql
  900.             raise original_exception

File ".virtualenvs/venv/lib/python3.5/site-packages/django/db/models/sql/compiler.py" in execute_sql
  890.             cursor.execute(sql, params)

File ".virtualenvs/venv/lib/python3.5/site-packages/django/db/backends/utils.py" in execute
  64.                 return self.cursor.execute(sql, params)

Exception Type: ValueError at /fckeditor/editor/filemanager/connectors/php/connector.php
Exception Value: A string literal cannot contain NUL (0x00) characters.
Request information:
USER: AnonymousUser
```

**Comment by gasman** [comment]
Thanks for the report, @benjaoming.

The reference to `/fckeditor/editor/filemanager/connectors/php/connector.php` confused me for far too long just now, but I guess that's (part of) the URL being requested by the malicious crawler...?

**Comment by jossingram** [comment]
We're currently also experiencing this on a daily basis, they're sending things like
/nph-mr.cgi?do=loginhelp&configLanguage=[Filtered]". Same error as above.

**Comment by acrewdson** [comment]
We're also seeing this, and even the 'Made with Wagtail' site might be susceptible: https://madewithwagtail.org/foo?bar=%00

I made a PR [here](https://github.com/wagtail/wagtail/pull/4652) – thanks for considering it.

cc: @chosak

**Comment by chosak** [comment]
Fixed by @acrewdson in #4652 and merged in 882f8f3.
